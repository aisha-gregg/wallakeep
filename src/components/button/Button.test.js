import React from "react";
import { Button } from "./Button";
import { render, fireEvent } from "@testing-library/react";

describe("Button", () => {
  it("should render the component", () => {
    const { container } = render(<Button></Button>);

    expect(container).toMatchSnapshot();
  });

  it("should render the component", () => {
    const mock = jest.fn();
    const { getByText } = render(<Button onClick={mock}>Test</Button>);
    const element = getByText("Test");

    fireEvent.click(element);

    expect(mock).toHaveBeenCalled();
  });
});
