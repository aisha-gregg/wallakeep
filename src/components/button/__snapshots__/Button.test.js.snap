// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the component 1`] = `
<div>
  <button
    class="button"
  />
</div>
`;
